### PR TITLE
Finish/Update API smoke tests

### LINKED ISSUE #1138: Add API smoke test
This should be a suite of tests to be run immediately after installing and configuring an instance. The requirements for this test are identical to the requirements outlined in #1056, but the interactions should take place through the API, not the CLI.

### PR DESCRIPTION
**Pull request by elyezer**
* Finish test_smoke test as now all entities are provided by NailGun.
  Closes #1138.
* Update test_end_to_end to use create and read methods instead of the
  create_json and read_json. By using those methods now is not needed to
  instantiate a new entity for every operation, the created entity can
  be reused. Also update the client section to make code cleaner.

Test results:

```
$ py.test tests/foreman/smoke/test_api_smoke.py -ktest_smoke
================================== test session starts ===================================
platform darwin -- Python 2.7.6 -- py-1.4.26 -- pytest-2.6.4
collected 8 items

tests/foreman/smoke/test_api_smoke.py .

========================== 7 tests deselected by '-ktest_smoke' ==========================
======================== 1 passed, 7 deselected in 296.90 seconds ========================

$ py.test tests/foreman/smoke/test_api_smoke.py -ktest_end_to_end
================================== test session starts ==================================
platform darwin -- Python 2.7.6 -- py-1.4.26 -- pytest-2.6.4
collected 8 items

tests/foreman/smoke/test_api_smoke.py .

======================= 7 tests deselected by '-ktest_end_to_end' =======================
======================= 1 passed, 7 deselected in 669.12 seconds ========================
```

### CODE REVIEW

File: tests/foreman/smoke/test_api_smoke.py [hunk 1]
```diff
@@ -836,162 +835,126 @@ def test_smoke(self):
...
-                {u'ids': [repo['id']], u'organization_id': org['id']},
-                auth=get_server_credentials(),
-                verify=False,
-            ).json()
-            self.assertGreater(
-                len(response['id']),
-                1,
-                u'Was not able to fetch a task ID.')
-            entities.ForemanTask(id=response['id']).poll()
+            repo.sync()
```
Comment (Ichimonji10) [inline tests/foreman/smoke/test_api_smoke.py]: :+1:

File: tests/foreman/smoke/test_api_smoke.py [hunk 2]
```diff
@@ -836,162 +835,126 @@ def test_smoke(self):
...
+        self.assertEqual(len(versions[0]['environment_ids']), 3)
 
         # BONUS: Create a content host and associate it with promoted
         # content view and last lifecycle where it exists
         content_host = entities.System(
-            content_view=content_view['id'],
-            environment=le2['id']
+            server_config,
+            content_view=content_view,
+            environment=le2.id  # FIXME: pass the environment instance
```
Comment (Ichimonji10) [inline tests/foreman/smoke/test_api_smoke.py]: Oh no! Is this broken?

**Comment by Ichimonji10** [comment]
ACK

File: tests/foreman/smoke/test_api_smoke.py (hunk 2 above)
Comment (omaciel) [inline tests/foreman/smoke/test_api_smoke.py]: @elyezer @Ichimonji10 is it?

Comment (elyezer) [inline tests/foreman/smoke/test_api_smoke.py]: Yes it is.

**Comment by elyezer** [comment]
Rebased.

**Comment by elyezer** [comment]
Rebased

**Comment by JacobCallahan** [comment]
ACK

**Pull request merged**